Bonsai.Shaders/BindTexture.cs

```diff
@@ -136,7 +136,12 @@ public IObservable<Texture> Process(IObservable<Texture> source)
             return Process(source, input =>
             {
                 GL.ActiveTexture(TextureSlot);
-                GL.BindTexture(TextureTarget, input != null ? input.Id : 0);
+                int textureId;
+                if (Index is int index)
+                    textureId = ((TextureSequence)input).Textures[index];
+                else
+                    textureId = input.Id;
```

Comment:
```suggestion
                var textureId = Index is int index
                    ? ((TextureSequence)input).Textures[index]
                    : input.Id;
```

This is great, for some reason it never hit that you could use nullables against value pattern matching, but of course it makes complete sense.

For consistency with the code above I would maybe just change this to ternary operator form. In fact you could probably also change the code in line 82 to use value pattern matching as it should be more efficient.

Reply:
Yup! Pattern matching is the best.

And done!